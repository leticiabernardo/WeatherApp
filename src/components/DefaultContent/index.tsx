import { Text, Box } from '@chakra-ui/react';
import LocationGif from '@/assets/images/location.gif';
import { useTranslation } from 'react-i18next';

const DefaultContent = () => {
  const { t } = useTranslation();

  return (
    <Box
      display="flex"
      alignItems="center"
      justifyContent="center"
      flexDirection="column"
      height="80%"
      width="100%"
      margin="30px 0"
      background="#222"
      borderRadius="20px"
    >
      <Box>
        <img src={LocationGif} width="140" alt="" />
      </Box>
      <Text color="#fff" fontSize="1xl" marginTop="30px" fontFamily="Open Sans">
        {t('Search to get started...')}
      </Text>
    </Box>
  );
};

export default DefaultContent;
