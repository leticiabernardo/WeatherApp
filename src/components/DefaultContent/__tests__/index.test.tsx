import { render, cleanup } from '@testing-library/react';
import DefaultContent from '../index';

describe('default content render', () => {
  afterEach(cleanup);

  test('render the component', () => {
    const { container } = render(<DefaultContent />);
    expect(container).toMatchSnapshot();
  });

  test('should be able to test component', () => {
    const { getByText } = render(<DefaultContent />);
    expect(getByText('Search to get started...')).toBeDefined();
    expect(() => getByText('no exist')).toThrow();
  });
});
