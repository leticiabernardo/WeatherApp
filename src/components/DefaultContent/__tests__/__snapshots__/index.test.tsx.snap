// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default content render render the component 1`] = `
<div>
  <div
    class="css-gnvv8y"
  >
    <div
      class="css-0"
    >
      <img
        alt=""
        src="[object Object]"
        width="140"
      />
    </div>
    <p
      class="chakra-text css-9sexf0"
    >
      Search to get started...
    </p>
  </div>
</div>
`;
